ui-tests/src/test/java/org/openmrs/reference/VisitNoteTest.java

```diff
@@ -51,25 +49,31 @@ public void VisitNoteTest() throws Exception {
         visitNotePage.addSecondaryDiagnosis(DIAGNOSIS_SECONDARY);
         visitNotePage.addNote("This is a note");
         patientDashboardPage = visitNotePage.save();
-        assertEquals(DIAGNOSIS_PRIMARY, visitNotePage.primaryDiagnosis());
-        assertEquals(DIAGNOSIS_SECONDARY, visitNotePage.secondaryDiagnosis());
-
-        PatientVisitsDashboardPage patientVisitsDashboardPage = patientDashboardPage.goToRecentVisits();
-
+        patientDashboardPage.waitForPage();
+        patientVisitsDashboardPage = patientDashboardPage.goToRecentVisits();
         EditVisitNotePage editVisitNotePage = patientVisitsDashboardPage.goToEditVisitNote();
-
-        editVisitNotePage.deleteDiagnosis();
+        
+        //Editing visit Note 
+        editVisitNotePage.waitForPage();
+        editVisitNotePage.removeDiagnosis();
+        editVisitNotePage.clearNote();
+        editVisitNotePage.addNote("This is edited note");
         editVisitNotePage.addSecondaryDiagnosis(DIAGNOSIS_SECONDARY_UPDATED);
-
-        //TODO Edit function doesn't work int02
-        patientDashboardPage = editVisitNotePage.save();
         assertEquals(DIAGNOSIS_SECONDARY_UPDATED, patientDashboardPage.secondaryDiagnosis());
```

Comment:
Shouldn't we have some more assertions to ensure the note is actually updated?

Reply:
Resolved thanks